src/main/java/com/liferay/faces/util/i18n/internal/I18nImpl.java

```diff
@@ -53,10 +55,27 @@ public I18nImpl() {
 
 		// Store the resource bundle cache in the application map (as a Servlet Context attribute).
 		if (startupFacesContext != null) {
+
 			ExternalContext externalContext = startupFacesContext.getExternalContext();
 			Map<String, Object> applicationMap = externalContext.getApplicationMap();
-			Map<Locale, ResourceBundle> cache = new ConcurrentHashMap<Locale, ResourceBundle>();
-			applicationMap.put(I18nImpl.class.getName(), cache);
+			Cache<Locale, ResourceBundle> facesResourceBundleCache;
+
+			String maxCacheCapacityString = externalContext.getInitParameter(I18n.class.getName() +
+					".maxCacheCapacity");
```

Comment:
@stiemannkj1 In the case in which it is unset (not specified in web.xml) then the default value is returned. You could specify -1 as a default value in WebConfigParam, indicating no maximum size was specified.